docs: add "Verifying a Rebase" section to CONTRIBUTING.md

PastaPastaPasta opened the pull request.

PastaPastaPasta commented:
force pushed to resolve lint failure

akshaynexus commented:
Doesn't the commit history shown ina pr do that already?ie when force pushed you can see what was changed in that force push

PastaPastaPasta commented:
@akshaynexus Kindof, but not really. If you look at https://github.com/dashpay/dash/pull/3415, you can see I rebased it on latest develop on the 12th. If you look at the change between force-pushes here https://github.com/dashpay/dash/compare/82b56774f57887867bd80db43ab2e0394e045b0d..bd3e07c3ea3c7b5f527aaccd637786cb7216a1a8 you'll see ALL the changes in develop between what I was based on, and what I rebased on.

It's very possible I could hide changes in there, so it's important to be able to check that the force-push doesn't really change anything which is what the gfd function does.

Does that make sense?
